src/libsphinxbase/feat/cmn_adapt.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * 16-Feb-2016  Zamir Ostroukhov <zamiron@gmail.com>
+ */
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#ifdef _MSC_VER
+#pragma warning (disable: 4244)
+#endif
+
+#include "sphinxbase/ckd_alloc.h"
+#include "sphinxbase/cmn.h"
+
+void
+cmn_adapt(cmn_t *cmn, mfcc_t **incep, int32 varnorm, int32 nfr) {
+
+    int32 i, j;
+    mfcc_t cep_cur[cmn->veclen];        // check it please, is it correctly?
```

Comment:
You can not allocate arrays in plan C, it's better to preallocate it like cep_max